index.bs

```diff
@@ -7546,9 +7547,19 @@ To <dfn export>run WebDriver BiDi preload scripts</dfn> given |environment setti
 
     1. If |preload script|'s <code>contexts</code> is not null:
 
-       1. Let |navigable id| be |navigable|’s [=navigable/top-level traversable=]'s id.
+       1. Let |is script run allowed| be false.
 
-       1. If |preload script|'s <code>contexts</code> does not [=list/contains|contain=] |navigable id|, continue.
+       1. While |navigable| is not null:
+
+          1. Let |navigable id| be |navigable|’s [=navigable/top-level traversable=]'s id.
+
+          1. If |preload script|'s <code>contexts</code> [=list/contains=] |navigable id|,
+            set |is script run allowed| to true and break.
+
+          1. If |preload script|'s <code>applyToOpeningContexts</code> is true,
```

Comment:
As a small detail, I'd prefer this to be `1. If |inherit| is false then break. 1. Set |navigable to |navigable|'s original opener`.

But I also feel like this doesn't quite capture the invariant that each preload script is defined exactly once so has a single setting for whether it's inherited. So conceptually if we're not inheriting we're compaing against a set containing a single value, whereas if we are inheriting we're comparing against a set that contains all the values from openers. I think making that structure explicit would make the spec easier to read.

As a different thought: if this applies to preload scripts it should also apply to event subscriptions in the same way, right? Even if we don't do it in the same PR, it could be worth checking if the same logic works in that case, or if we want to hook in to the context creation lifecycle and update the underlying datastructures to explicitly add the new context, so the getters are still doing the same lookup.

Reply:
We would need to apply it to the event subscriptions and, based on the current feedback from the clients, to the viewport configuration.

Reply:
Yes, basically anything that's understood as browser configuration rather than interacting with a specific document.